source/dofs/dof_handler.cc

```diff
@@ -250,6 +250,28 @@ namespace internal
         return std::min(max_couplings, dof_handler.n_dofs());
       }
 
+      /**
+       * Do that part of reserving space that pertains to releasing
+       * the previously used memory.
+       */
+      template <int dim, int spacedim>
+      static void
+      reserve_space_release_space(DoFHandler<dim, spacedim> &dof_handler)
+      {
+        dof_handler.object_dof_indices.assign(dof_handler.tria->n_levels(), {});
+        dof_handler.object_dof_indices.shrink_to_fit();
```

Comment:
I don't which is faster: either replacing all entries of this vector with empty containers, or iterating over all entries and calling `clear` on each container. Anyways, I find this way easier to read!

Reply:
Yeah, that does the job, too.

Reply:
I will leave the `shrink_to_fit` call to also adjust the capacity. Alternatively, we can also `swap` with an empty vector.